lib/iris/analysis/__init__.py

```diff
@@ -186,15 +185,26 @@ def matches_any(self, predicate):
         return any(self.matches(predicate))
 
 
-def coord_comparison(*cubes, object_get=None):
+def _dimensional_metadata_comparison(*cubes, object_get=None):
     """
-    Convenience function to help compare coordinates on one or more cubes
-    by their metadata.
+    Convenience function to help compare coordinates, cell-measures or
+    ancillary-variables, on one or more cubes, by their metadata.
 
     Return a dictionary where the key represents the statement,
-    "Given these cubes list the coordinates which,
+    "Given these cubes list the elements which,
     when grouped by metadata, are/have..."
 
+    .. Note::
```

Comment:
Placing this note between: `when grouped by metadata, are/have..."` and `Keys:` somewhat breaks up the flow of the docstring the `are/have` relates to the keys listed below. This note might be better placed elsewhere.

Reply:
Well I didn't want to leave it until after this long long long list of keys in the returned data
-- Not least because, that info should now  be read in the light of this note (i.e. it assumes everything is a coord).

Also, none of this gets rendered, now it is private, so I don't think style is such a key factor.
( The existing is not in full alignment with the style anyway, I reckon )